ext/drmgr/drmgr.c

```diff
@@ -2058,6 +2049,62 @@ drmgr_unregister_thread_exit_event_user_data(void (*func)(void *drcontext,
                                       (void (*)(void))func);
 }
 
+DR_EXPORT
+bool
+drmgr_register_filter_syscall_event(bool (*func)(void *drcontext, int sysnum))
+{
+    return drmgr_generic_event_add(&cblist_filter_sys, filter_sys_event_lock,
```

Comment:
Why do all drmgr APIs have with and without user_data versions (the user can just pass a null perhaps)? Is the "without" version expected to be simpler to use? Or maybe we added the "without" versions in the past, and now we want them for all future ones also?